@utahdts/utah-design-system/react/components/containers/tabs/TabGroup.jsx

```diff
@@ -18,14 +19,25 @@ export function TabGroup({
   children,
   className,
   defaultValue,
+  isVertical,
   onChange,
   value,
 }) {
   const tabGroupId = useId();
+  const tabGroupRef = useRef(/** @type {HTMLDivElement | null} */(null));
   const [tabGroupState, setTabGroupState] = useImmer(() => ({
     selectedTabId: defaultValue,
     tabGroupId,
+    tabs: /** @type {NodeList | []} */ [],
   }));
+  const navigateTab = useCallback((/** @type {HTMLButtonElement | null} */ tab) => {
+    if (tab) {
+      tab?.focus();
+      tab?.click();
+    }
+  }, []);
+  const findCurrentTabIndex = useCallback(() => (tabGroupState.tabs.findIndex((/** @type {HTMLButtonElement | null} */ tab) => tab?.id === `tab-${tabGroupState.tabGroupId}-${tabGroupState.selectedTabId}`)), [tabGroupState]);
```

Comment:
This is scary because if the Tab's id formation code is ever changed, this will break trying to find a tab. Maybe a comment in both places that they are "copy-paste" code would be helpful to reminder a future developer to keep the two locations in sync? Or maybe make a `tabId()` function that both places call?

Another option would be to track the current tab index instead of all the tabs. The focus() and click() in `navigateTab` would need to be moved to a useEffect in Tab:
```
useEffect(
() => { 
// myTabIndex could be returned from `myTabIndexRef.current = registerTab()`
if (tabIndexInContext === myTabIndexRef.current) { 
myRef.current.focus();
myRef.current.click();
}
},
[tabIndexInContext]
);
```